solutions/gold/cc-KSIZEGCD.mdx

```diff
@@ -13,9 +13,9 @@ Can we upper bound the number of times the GCD changes?
 
 <Spoiler title="Answer to Hint 1">
 
-Note that the GCD can only decrease, and it will always remove some factor from the current GCD.
-The smallest factor we can remove is $2$, and $a_i \leq 10^9$, so the GCD can only change at most
-$log_2(10^9) \approx 30$ times.
+Note that the GCD can only decrease and there are two ways for it to do so, it will either remove some factor from the current GCD or find another GCD at a different position.
```

Comment:
hm, i think the wording of this editorial is a little confusing -- it should be "the GCD can only decrease **as you extend the subarray**". sorry about that!

i'll fix it asap 👍 

let me know if you have any more questions/comments!

Reply:
What I meant is that you could have to recalculate the GCD up to 446 times due to the constraints of the problem  (sorry English isn't my first language)

Reply:
Imagine an Array like this: [A, 1, A-1, A-1, 1, A-2, A-2, A-2, 1...] this means the gcd change x times where x is: (x+2)*(x-1)/2 < N if N is 100000 the greatest x that satisfy the formula is 446.

P.S. it is also possible that the array both decreases by finding another gcd at a different position but also by removing some factor, so it can change more or less 446 + log(N) times ≈ 470/480

Reply:
the $\log(10^9)$ about extending a singular subarray, so i don't think this would work.

in your array, the gcd would change like this:

gcd of [A] = A
gcd of [A, 1] = 1
gcd of [A, 1, A - 1] = 1
gcd of [A, 1, A - 1, ...] = 1

---

im not too sure what you mean by finding another gcd at a different position ?

Reply:
you need the greatest gcd for every subarray of length x where x <= N, for x = 1 you have gcd = A (subarray [:1]), for x = 2 you have gcd = A-1 (subarray [2:5]) ecc... The solution for this problem recalculates the gcd in O(n) time every time the gcd decreases. I don't understand why you are always starting from the first element of the array. Also I already tested this, try an input like that whit this code: https://pastebin.com/9BsGMANF you will se that it outputs 446.

Reply:
I now understand why you are starting always from the start, the code in the solution does that, but in the hint it should specify that the gcd can change at maximum log(10^9) for every subarray starting at a specific position, otherwise it seems that it can change a maximum of log(10^9) times in the answer.

Reply:
yep! that's what I said in my first comment (about the confusing wording).

sorry for the confusion -- if everything is fine, ill make the changes and close this pr. does that sound alright to you?

Reply:
yes, perfect.

Sorry I didn't understand initially